src/lib.rs

```diff
@@ -130,7 +130,7 @@ impl<'a, R: Rng> MarkovChain<'a, R> {
         }
         // Sync the keys with the current map.
         self.keys = self.map.keys().cloned().collect();
-        self.keys.sort();
+        self.keys.sort_unstable();
```

Comment:
I believe this is unrelated to the update, so please submit this in a separate PR, thanks! Then you can explain there why this is a great change :smile: